hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorState.java

```diff
@@ -73,23 +73,32 @@
 
     /**
      * Making calls to {@link Processor#snapshotCommitFinish(boolean)} until it
-     * returns {@code true}.
+     * returns {@code true} and then return to {@link #PROCESS_INBOX}. Used
+     * when phase-2 was initiated while in {@link #PROCESS_INBOX}.
      */
-    ON_SNAPSHOT_COMPLETED,
+    SNAPSHOT_COMMIT_FINISH__PROCESS,
 
     /**
-     * Processor completed after a phase 1 of a snapshot and is not doing
-     * anything, but it cannot be done until a phase 2 is done - this state is
-     * waiting for a phase 2.
+     * Making calls to {@link Processor#snapshotCommitFinish(boolean)} until it
+     * returns {@code true} and then return to {@link #COMPLETE}. Used when
+     * phase-2 was initiated while in {@link #COMPLETE}.
      */
-    WAITING_FOR_SNAPSHOT_COMPLETED,
+    SNAPSHOT_COMMIT_FINISH__COMPLETE,
+
+    /**
+     * Making calls to {@link Processor#snapshotCommitFinish(boolean)} until it
+     * returns {@code true} and then proceed to {@link #EMIT_DONE_ITEM}. Used
+     * when phase-2 was initiated after {@code Processor.complete()} returned
+     * true.
+     */
+    SNAPSHOT_COMMIT_FINISH__FINAL,
```

Comment:
is the double underscore intentional here? looks a little strange

Reply:
yes, there are three constants starting with `SNAPSHOT_COMMIT_FINISH`, after the two underscores is the discriminating part, all three have `__`.

Checkstyle didn't object :)